src/plugins/intel_cpu/src/nodes/kernels/rdft_kernel.cpp

```diff
@@ -393,50 +365,43 @@ void jit_dft_kernel_f32<isa>::generate() {
     this->postamble();
 }
 
+// Interleave real and imag registers and store in memory.
+// For example (for AVX):
+// real = [1, 2, 3, 4, 5, 6, 7, 8]
+// imag = [11, 12, 13, 14, 15, 16, 17, 18]
+// interleaved = [1, 11, 2, 12, 3, 13, 4, 14, 5, 15, 6, 16, 7, 17, 8, 18]
 template <cpu_isa_t isa>
-void jit_dft_kernel_f32<isa>::uni_vbroadcastsd(const Xbyak::Xmm& x, const Xbyak::Operand& op) {
-    movsd(x, op);
-    shufpd(x, x, 0x0);
+void jit_dft_kernel_f32<isa>::interleave_and_store(const Xbyak::Zmm& real, const Xbyak::Zmm& imag, const Xbyak::RegExp& reg_exp, const Xbyak::Zmm& tmp) {
```

Comment:
That is not right way to overload such function. It will be compiled for AVX2 and SSE as well, but they do not support ZMM register and `vpermt2ps` instruction.
Another two functions `interleave_and_store` also will be compiled for all isa. So, in total there will be 9 compiled instances of this function, but in current implementation just 3 of them will be used. 
Even assuming that usage of this function will be changed and for example on AVX512 it will be called with YMM or XMM registers, then not optimal implementation will be called.
It's better to declare one function with XMM args and define instances for each ISA.

Reply:
done